src/main/kotlin/biokotlin/kmer/KmerIO.kt

```diff
@@ -0,0 +1,225 @@
+package biokotlin.kmer
+
+import it.unimi.dsi.fastutil.BigArrays
+import net.jpountz.lz4.LZ4FrameInputStream
+import net.jpountz.lz4.LZ4FrameOutputStream
+import java.io.*
+
+/**
+ * This class reads kmers from a text file (which may be LZ4 compressed)
+ * and can be used to construct KmerSet, KmerBigSet, and KmerMultiSet objects.
+ * It can also iterate through a kmer text file without constructing a set.
+ */
+class KmerIO(filename: String, isCompressed: Boolean = true): Iterator<Pair<Kmer, Int>> {
+    private val reader: BufferedReader
+
+    private var sequenceLength: Long? = null
+    private var ambiguousKmers: Long? = null
+    private var setType: String? = null
+    private var stepSize: Int? = null
+    private var keepMinOnly: Boolean? = null
+    private var bothStrands: Boolean? = null
+    private var kmerSize: Int? = null
+    private var currentLine: String? = ""
+
+    init {
+        assert(filename.isNotEmpty())
+
+        reader = if (isCompressed) {
+            BufferedReader(InputStreamReader(LZ4FrameInputStream(FileInputStream(File(filename)))))
+        } else {
+            File(filename).bufferedReader()
+        }
+
+        // first line should be a header
+        val firstLine = reader.readLine()
+        if(!firstLine.startsWith("#")) { throw FileNotFoundException("File does not contain a header line")}
```

Comment:
This is a bit misleading as the file is found.  The IllegalFormat* exceptions seem to imply problems with specific datapoints, so perhaps IllegalStateException, and an expanded comment as what the header should look like?